doc/api/json-handlers-full-keyboard-lighting.md

```diff
@@ -0,0 +1,44 @@
+# JSON handlers for Full-Keyboard Lighting Effects
+
+Note: This document describes the specific handler format for full-keyboard effect handlers.  See [Writing Handlers in JSON][json-handlers] for a general overview of how to bind handlers.  See [JSON Color Handlers][json-handlers-color] for specifics on color and lighting handlers for specific keys and zones.
+
+Bitmap-type handlers are used to simultaneously control each key on RGB per-key-illuminated keyboards.
+
+## Full schema definition ##
+
+Each portion is described in detail in later sections.
+
+Top-level schema
+
+```
+`device-type`: 'rgb-per-key-zones'            mandatory.  This type of lighting is currently only supported for the rgb-per-key-zones device type.
+`mode`: `bitmap` | `partial-bitmap`           mandatory
+`excluded-events`: [<event name string>...]   optional, only supported for partial-bitmap.
+```
+
+## Full keyboard effects ##
+
+The `bitmap` visualization mode is a special case that can be used to individually control an entire keyboard's lighting each update.  To use this mode, your event data must include a context frame data key named `bitmap`.  This key's value must be a 132-length array of colors, each of which should be a 3-length array specifying R, G, and B values.  This array is interpreted as a 22x6 grid that is automatically mapped to the nearest appropriate keys on the user's keyboard.   Any parts of the array that do not map to a key on the user's keyboard are ignored.
```

Comment:
* The bitmap origin should be documented (top-left)
* There should be an example payload JSON, or at least a truncated one.